WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubTimeRangeGenerator.swift

```diff
@@ -0,0 +1,208 @@
+import Foundation
+import Yosemite
+
+struct AnalyticsHubTimeRange {
+    let start: Date
+    let end: Date
+}
+
+/// Main source of time ranges of the Analytics Hub, responsible for generating the current and previous dates
+/// for a given Date and range Type alongside their UI descriptions
+///
+public class AnalyticsHubTimeRangeGenerator {
+
+    private let currentTimezone = TimeZone.autoupdatingCurrent
+    private let currentDate: Date
+    private let currentCalendar: Calendar
+    private let selectionType: SelectionType
+
+    private var _currentTimeRange: AnalyticsHubTimeRange? = nil
+    private var _previousTimeRange: AnalyticsHubTimeRange? = nil
+
+    var currentTimeRange: AnalyticsHubTimeRange {
```

Comment:
Since these methods are meant to be the public API, A quick documentation for them would be useful!

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/8193/commits/9c7fed19e7e918780f97399c269c568f4867d063 and https://github.com/woocommerce/woocommerce-ios/pull/8193/commits/f23614a63440b92e1b0a1622e6165441abe2dcf0